sdk/core/azure-core/src/main/java/com/azure/core/util/DateTimeRfc1123.java

```diff
@@ -47,6 +48,23 @@ public OffsetDateTime getDateTime() {
         return this.dateTime;
     }
 
+    /**
+     * JSON creator for DateTimeRfc1123.
+     * <p>
+     * If {@code date} is null or an empty string null will be returned.
+     *
+     * @param date RFC1123 datetime string.
+     * @return The DateTimeRfc1123 representation of the datetime string, or null if {@code date} is null or empty.
+     */
+    @JsonCreator
+    static DateTimeRfc1123 fromString(final String date) {
+        if (CoreUtils.isNullOrEmpty(date)) {
+            return null;
+        }
```

Comment:
This is different behavior from the constructor that takes a string input. I guess the constructor throws an exception and we should have the same behavior here.

Reply:
That is correct that the behavior is different, in most cases empty string won't appear in the constructor given JSON handling is a bit better than XML (where `<XmlProperty></XmlProperty>` may be handled differently than `<XmlProperty/>`). Without this change the XmlMapper itself would need to have coercion configuration changes for DateTimeRfc1123 to treat empty string as null.

I think this change is more appropriate than having a coercion and matches the pattern we've seen more often with models in `azure-core` with JsonCreators.